postgresql/provider.go

```diff
@@ -246,6 +255,32 @@ func getRDSAuthToken(region string, profile string, username string, host string
 		return "", err
 	}
 
+	if role != "" {
+		stsClient := sts.NewFromConfig(awscfg)
+		roleInput := &sts.AssumeRoleInput{
+			RoleArn:         aws.String(role),
+			RoleSessionName: aws.String("TerraformSession"),
+		}
+
+		roleOutput, err := stsClient.AssumeRole(ctx, roleInput)
+		if err != nil {
+			return "", err
+		}
+
+		awscfg, err = awsConfig.LoadDefaultConfig(ctx,
+			awsConfig.WithCredentialsProvider(
+				aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(
+					*roleOutput.Credentials.AccessKeyId,
+					*roleOutput.Credentials.SecretAccessKey,
+					*roleOutput.Credentials.SessionToken,
+				)),
+			),
+		)
+		if err != nil {
+			return "", err
```

Comment:
liek previous comment
```suggestion
			return "", fmt.Errorf("could not load AWS default config", err)
```